src/Momento.Sdk/Internal/ScsTopicClient.cs

```diff
@@ -0,0 +1,183 @@
+            Value = value
+        };
+
+        try
+        {
+            _logger.LogTraceExecutingTopicRequest(RequestTypeTopicPublish, cacheName, topicName);
+            await grpcManager.Client.publish(request, new CallOptions(deadline: CalculateDeadline()));
+        }
+        catch (Exception e)
+        {
+            return _logger.LogTraceTopicRequestError(RequestTypeTopicPublish, cacheName, topicName,
+                new TopicPublishResponse.Error(_exceptionMapper.Convert(e)));
+        }
+
+        return _logger.LogTraceTopicRequestSuccess(RequestTypeTopicPublish, cacheName, topicName,
+            new TopicPublishResponse.Success());
+    }
+
+    private async Task<TopicSubscribeResponse> SendSubscribe(string cacheName, string topicName,
+        ulong? resumeAtTopicSequenceNumber)
+    {
+        var request = new _SubscriptionRequest
+        {
+            CacheName = cacheName,
+            Topic = topicName
+        };
+        if (resumeAtTopicSequenceNumber != null)
+        {
+            request.ResumeAtTopicSequenceNumber = resumeAtTopicSequenceNumber.Value;
+        }
+
+        AsyncServerStreamingCall<_SubscriptionItem> subscription;
+        try
+        {
+            _logger.LogTraceExecutingTopicRequest(RequestTypeTopicPublish, cacheName, topicName);
+            subscription = grpcManager.Client.subscribe(request, new CallOptions());
+        }
+        catch (Exception e)
+        {
+            return _logger.LogTraceTopicRequestError(RequestTypeTopicPublish, cacheName, topicName,
+                new TopicSubscribeResponse.Error(_exceptionMapper.Convert(e)));
+        }
+
+        var response = new TopicSubscribeResponse.Subscription(
+            token => MoveNextAsync(subscription, token, cacheName, topicName),
+            () => subscription.Dispose());
+        return _logger.LogTraceTopicRequestSuccess(RequestTypeTopicPublish, cacheName, topicName,
+            response);
+    }
+
+    private async ValueTask<TopicMessage?> MoveNextAsync(AsyncServerStreamingCall<_SubscriptionItem> subscription,
+        CancellationToken cancellationToken, string cacheName, string topicName)
+    {
+        if (cancellationToken.IsCancellationRequested)
+        {
+            return null;
+        }
+
+        try
+        {
+            while (await subscription.ResponseStream.MoveNext(cancellationToken).ConfigureAwait(false))
+            {
+                var message = subscription.ResponseStream.Current;
+
+                switch (message.KindCase)
+                {
+                    case _SubscriptionItem.KindOneofCase.Item:
+                        _logger.LogTraceTopicMessageReceived("item", cacheName, topicName);
+                        return new TopicMessage.Item(message.Item);
+                    case _SubscriptionItem.KindOneofCase.Discontinuity:
+                        _logger.LogTraceTopicMessageReceived("discontinuity", cacheName, topicName);
+                        break;
```

Comment:
this should probably be returning a subscription item so users can be aware of missing messages.

Reply:
I was going to add Discontinuity as a separate item in a second pass, but I can add it now if if won't add too much complexity to the pr.

Reply:
yes we need to get this out the door, we will add those later